src/triage/component/postmodeling/contrast/parameters.py

```diff
@@ -0,0 +1,71 @@
+"""
+Postmodeling parameters
+
+This script contain the parameters Class that will be used across all the
+postmodeling functions within the ModelEvaluator and ModelGroupEvaluator
+classes. This class will be initialized using the 'postmodeling_parameters.yaml' 
+file
+
+"""
+
+import yaml
+import json
+import warnings
+
+class  ThresholdIterator(dict):
```

Comment:
What you've defined **is** a dictionary -- **not** an iterator -- it extends `dict`.

Now it can be whatever you need it to be. But in its current state, it's a bit confusing.

Is it that you want to preserve the hash-key access of the dictionary parameter? If so, good -- keep it a `dict` (extension). I will, however, again attempt to dissuade you from defining a class which is in nearly every way a generic `dict`, except that it lies about its length (`__len__`) and contents (`__iter__`).

If you want to iterate over only your `elements` -- also good, because the constructor of this class allows you to do so -- `for (metric, threshold) in threshold_dict.elements: ...`. No need to pretend that the dictionary itself has the structure of the `elements` list it constructs.

If, on the other hand, all you really need is a utility to translate your threshold dictionary into an elements iterator -- then all you need is a very simple, straightforward function, in this case, a **generator**:

```
def generate_thresholds(threshold_dict):
    for (metric, param) in threshold_dict.items():
        for threshold in param:
            yield (metric, threshold)

for (metric, threshold) in generate_thresholds(thresholds):
    ...
```

If you want length as well (_i.e._ a proper list like `elements`):

```
def threshold_elements(threshold_dict):
    return [
        (metric, threshold)
        for (metric, param) in threshold_dict.items()
        for threshold in param
    ]

elements = threshold_elements(thresholds)
...
```